app/src/main/java/org/xbmc/kore/ui/sections/video/PVRRecordingsListFragment.java

```diff
@@ -134,10 +228,90 @@ public void onSuccess(List<PVRType.DetailsRecording> result) {
 
                 // To prevent the empty text from appearing on the first load, set it now
                 emptyView.setText(getString(R.string.no_recordings_found_refresh));
-                setupRecordingsGridview(result);
+
+                // As the JSON RPC API does not support sorting or filter parameters for PVR.GetRecordings
+                // we apply the sorting and filtering right here.
+                // See https://kodi.wiki/view/JSON-RPC_API/v9#PVR.GetRecordings
+                List<PVRType.DetailsRecording> finalResult = filter(result);
+                sort(finalResult);
+
+                setupRecordingsGridview(finalResult);
                 swipeRefreshLayout.setRefreshing(false);
             }
 
+            private List<PVRType.DetailsRecording> filter(List<PVRType.DetailsRecording> itemList) {
+                SharedPreferences preferences = PreferenceManager.getDefaultSharedPreferences(getActivity());
+                boolean hideWatched = preferences.getBoolean(Settings.KEY_PREF_PVR_RECORDINGS_FILTER_HIDE_WATCHED, Settings.DEFAULT_PREF_PVR_RECORDINGS_FILTER_HIDE_WATCHED);
+
+                boolean filter = hideWatched;
```

Comment:
Redundant var, can use `hideWatched`

Reply:
I've had in mind here that there may be additional conditions added later on - e.g. an outstanding PR for #695 would add a `boolean hasSearchFilter` condition.